ark/spLDA/processing.py

```diff
@@ -1,56 +1,45 @@
-import functools
-import matplotlib.pyplot as plt
-import numpy as np
 import pandas as pd
-import os
-import pickle
-from ark.utils.spatial_lda_utils import check_format_cell_table_args
-# from spatial_lda.featurization import *
-# from spatial_lda.visualization import plot_adjacency_graph()
+from sklearn.model_selection import train_test_split
+from spatial_lda.featurization import *
 
+from ark.settings import BASE_COLS
+from ark.utils.spatial_lda_utils import check_format_cell_table_args, \
+    check_featurize_cell_table_args
+
+train_frac = 0.75
+
+
+def format_cell_table(cell_table, markers=None, clusters=None):
+    """Formats a cell table containing one for more fields of view to be
+    compatible with the spatial_lda library.
 
-def format_cell_table(cell_table, markers=None, clusters=None, fovs="all_fovs"):
-    """
-    Formats a cell table containing one for more fields of view to be compatible with the spatial_lda library.
     Args:
         cell_table (pd.DataFrame):
-            A pandas DataFrame containing the columns of cell marker frequencies and/or cluster ids.
+            A pandas DataFrame containing the columns of cell marker
+            frequencies and/or cluster ids.
         markers (list, str):
-            A list of strings corresponding to marker names.
-            Default: None
+            A list of strings corresponding to marker names.  Either markers
+            or clusters must be provided.
         clusters (list, int):
             A list of integers corresponding to cluster ids.
-            Default: None
-        fovs (str, int):
-            One of either "all_fovs" indicating all field of views are to be kept, or a list of integers corresponding
-            to the index of each field of view which should be kept.
-            Default: "all_fovs"
 
     Returns:
-        pd.DataFrame:
-
-        - A formatted cell table for use in spatial-LDA analysis
+        dict:
 
+        - A dictionary of formatted cell tables for use in spatial-LDA
+        analysis.  Each element in the dictionary is a
+        pd.DataFrame corresponding to a single field of view.
     """
 
     # Check function arguments
-    check_format_cell_table_args(cell_table=cell_table, markers=markers, clusters=clusters, fovs=fovs)
+    check_format_cell_table_args(cell_table=cell_table, markers=markers,
+                                 clusters=clusters)
 
     # Only keep columns relevant for spatial-LDA
-    all_columns = cell_table.columns
-    keep_columns = [
-        "point",
-        "label",
-        "cell_size",
-        "centroid-0",
-        "centroid-1",
-        "pixelfreq_hclust_cap",
-        "name"
-    ]
     if markers is not None:
-        keep_columns.append(markers)
-    drop_columns = [c for c in all_columns if c not in keep_columns]
-    cell_table = cell_table.drop(columns = drop_columns)
+        BASE_COLS.append(markers)
+    drop_columns = [c for c in cell_table.columns if c not in BASE_COLS]
```

Comment:
You mean pixelfreq_hclust_cap? And yes it would keep it and also rename it as cluster_id.

Reply:
I was worried this would mean that the cluster_freqs were getting included in the spatial lda computation, but that's not the case because the featurization function explicitly takes the list of markers. Given that, what is the purpose of subsetting this cell_table to only include the relevant markers if the featurization function already does that? Just memory reason?